cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -128,7 +131,9 @@ public KafkaBrokerConfigurationBuilder withCruiseControl(String clusterName, Cru
                                                              String replicationFactor, String minInSyncReplicas)   {
         if (cruiseControl != null) {
             printSectionHeader("Cruise Control configuration");
-            writer.println(CruiseControlConfigurationParameters.METRICS_TOPIC_NAME + "=strimzi.cruisecontrol.metrics");
+            Map<String, Object> config = Optional.ofNullable(cruiseControl.getConfig()).orElseGet(HashMap::new);
+            String metricsTopicName = config.getOrDefault(CruiseControlConfigurationParameters.CRUISE_CONTROL_METRIC_REPORTER_TOPIC_NAME.getValue(), CruiseControlConfigurationParameters.DEFAULT_METRIC_REPORTER_TOPIC_NAME).toString();
+            writer.println(CruiseControlConfigurationParameters.METRICS_TOPIC_NAME + "=" + metricsTopicName);
```

Comment:
I wonder if using `if` instead of creating a new map would be better from memory perspective. Or maybe we can at least use `Map.of()` or something to create small unmodifiable map?

Reply:
I wondered about:
```
String metricsTopicName = Optional.ofNullable(cruiseControl.getConfig())
                    .map(config -> config.get(CruiseControlConfigurationParameters.CRUISE_CONTROL_METRIC_REPORTER_TOPIC_NAME.getValue()))
                    .map(Object::toString)
                    .orElse(CruiseControlConfigurationParameters.DEFAULT_METRIC_REPORTER_TOPIC_NAME);
```

but didn't know if that would also be less readable?

Reply:
It is. But maybe if it saves memory there is at least a reason for it. Do the two `.map()` after each other work? Doesn't that throw NPE?

Reply:
I don't think it throws an NPE, my understanding is that the `.map()` is clever enough that if any return null it would drop down to the `.orElse()`